### DISCUSSION TITLE
Can I override the printer for `std::string`?

### DISCUSSION THREAD

**Discussion by ChrisThrasher** [onqtam/doctest#764] [Q&A]
I'm working on some string encoding code that returns a `std::string` that was converted from a `std::wstring`. This often produces a `std::string` with unprintable characters. I'd like to print the hex code for these to avoid seeing a question mark in the console when a test fail. However it seems like I can't override the stringification of `std::string`. Is there any way I can override that?

**Comment by onqtam** [comment]
Not sure (and this is strange - coming from the author of the library :D )
Take a look at the stringification docs: https://github.com/doctest/doctest/blob/master/doc/markdown/stringification.md
Perhaps using a doctest::StringMaker<T> specialisation?
